### PR TITLE
[OSDOCS-2969]: Azure Marketplace images in machineset YAML

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.8+

Also 4.8-4.10, but needs to wait for the installer content backport, which is not a simple CP. ~Will add labels and CP for 4.8-4.10 later.~
Cc: @bscott-rh

Installer backports:
4.8: https://github.com/openshift/openshift-docs/pull/50828
4.9: https://github.com/openshift/openshift-docs/pull/50895
4.10: https://github.com/openshift/openshift-docs/pull/51370

Issue:
[OSDOCS-2969](https://issues.redhat.com//browse/OSDOCS-2969)

Link to docs preview:
- [Sample YAML for a machine set custom resource on Azure
](http://file.rdu.redhat.com/jrouth/OSDOCS-2969-Azure-marketplace-machine-sets/machine_management/creating_machinesets/creating-machineset-azure.html#machineset-yaml-azure_creating-machineset-azure)
- [Selecting an Azure Marketplace image](http://file.rdu.redhat.com/jrouth/OSDOCS-2969-Azure-marketplace-machine-sets/machine_management/creating_machinesets/creating-machineset-azure.html#installation-azure-marketplace-subscribe_creating-machineset-azure)
- [Sample YAML for a machine set custom resource on Azure](http://file.rdu.redhat.com/jrouth/OSDOCS-2969-Azure-marketplace-machine-sets/machine_management/creating-infrastructure-machinesets.html#machineset-yaml-azure_creating-infrastructure-machinesets) (infra version)

Additional information:
Relates to #48577 (installer, should merge first) and #48753 (AWS machine set content)

### CODE REVIEW

**Review by JoelSpeed — COMMENTED** [review]
From a process perspective looks good from what I remember

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by jeana-redhat** [comment]
Can one of @sunzhaohua2, @miyadav, or @huali9 please take a look at this PR when you have time?

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+endif::ipi[]
+ifdef::mapi[]
+. Add the following details to the `providerSpec` section of your machine set YAML file:
++
+.Sample `providerSpec` image values for Azure Marketplace compute machines
+
+[source,yaml]
+----
+providerSpec:
+        value:
```
Comment (sunzhaohua2) [inline modules/installation-azure-marketplace-subscribe.adoc:158]: nit, indentation is not correct

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+          image:
+            offer: rh-ocp-worker
+            publisher: redhat
+            resourceID: /resourceGroups/<infrastructure_id>-rg/providers/Microsoft.Compute/images/<infrastructure_id> <1>
+            sku: rh-ocp-worker
+            version: 4.8.2021122100
```
Comment (sunzhaohua2) [inline modules/installation-azure-marketplace-subscribe.adoc:164]: this should be like below, we need to unset the `resourceID`, add `type: MarketplaceWithPlan`
```
          image:
            offer: rh-ocp-worker
            publisher: redhat
            resourceID: ""
            sku: rh-ocp-worker
            type: MarketplaceWithPlan
            version: 4.8.2021122100
```

Comment (jeana-redhat) [reply modules/installation-azure-marketplace-subscribe.adoc]: Thanks Zhaohua! I have updated to match your snippet. 

Just to make sure I understand: I have also removed the following note about the `resourceID`, is that correct?
```
<1> Specify an image that is compatible with your instance type. The Hyper-V generation V2 images created by the installation program have a `-gen2` suffix, while V1 images have the same name without the suffix.
```

Comment (sunzhaohua2) [reply modules/installation-azure-marketplace-subscribe.adoc]: Yes, it is correct.

Comment (jeana-redhat) [reply modules/installation-azure-marketplace-subscribe.adoc]: Thank you, as always!

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 3]
```diff
@@ -133,10 +146,31 @@ compute:
   replicas: 3
 ----
 endif::ipi[]
+ifdef::mapi[]
+. Add the following details to the `providerSpec` section of your machine set YAML file:
```
Comment (bscott-rh) [inline modules/installation-azure-marketplace-subscribe.adoc:150]:
```suggestion
. Add the following parameters to the `providerSpec` section of your machine set YAML file using the image details of your offer:
```
If the user is supposed to populate this sample code with the values from the previous step, you might want to either mention it in this step, or replace the values in the code block with `<value_that_you_should_replace>` type of wording. What do you think?

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 4]
```diff
@@ -2,20 +2,30 @@
...
+//jrouth: Also some variations for when it appears in the machine management content (`mapi`).
 
 :_content-type: PROCEDURE
 [id="installation-azure-marketplace-subscribe_{context}"]
 = Selecting an Azure Marketplace image
+ifndef::mapi[]
 If you are deploying an {product-title} cluster using the Azure Marketplace offering, you must first obtain the Azure Marketplace image. The installation program uses this image to deploy worker nodes. When obtaining your image, consider the following:
+endif::mapi[]
+ifdef::mapi[]
+You can create a machine set running on Azure that deploys machines using the Azure Marketplace offering. To use this offering, you must first obtain the Azure Marketplace image. When obtaining your image, consider the following:
```
Comment (adellape) [inline modules/installation-azure-marketplace-subscribe.adoc:27]: s/using/by using/

? And if so, line 24 as well.

File: modules/installation-azure-marketplace-subscribe.adoc [hunk 5]
```diff
@@ -133,10 +146,31 @@ compute:
   replicas: 3
 ----
 endif::ipi[]
+ifdef::mapi[]
+. Add the following parameters to the `providerSpec` section of your machine set YAML file using the image details for your offer:
++
+.Sample `providerSpec` image values for Azure Marketplace compute machines
+
```
Comment (adellape) [inline modules/installation-azure-marketplace-subscribe.adoc:153]: Renders fine as-is, but maybe remove blank line here.

File: modules/installation-azure-marketplace-subscribe.adoc (hunk 4 above)
Comment (jeana-redhat) [reply modules/installation-azure-marketplace-subscribe.adoc:27]: actually "that use" - but very good point, it isn't clear :nerd_face:

**Pull request merged**